pkg/csi_driver/controller.go

```diff
@@ -524,34 +548,24 @@ func invalidCapacityRange(capRange *csi.CapacityRange, tier string, validRange *
 	return nil
 }
 
-// init function to get min and max volume sizes per tier
-func provisionableCapacityForTier(tier string) *capacityRangeForTier {
-	defaultRange := capacityRangeForTier{min: defaultTierMinSize, max: defaultTierMaxSize}
-	enterpriseRange := capacityRangeForTier{min: enterpriseTierMinSize, max: enterpriseTierMaxSize}
-	highScaleRange := capacityRangeForTier{min: highScaleTierMinSize, max: highScaleTierMaxSize}
-	premiumRange := capacityRangeForTier{min: premiumTierMinSize, max: premiumTierMaxSize}
-	zonalRange := capacityRangeForTier{min: zonalTierMinSize, max: zonalTierMaxSize}
-	provisionableCapacityForTier := map[string]capacityRangeForTier{
-		defaultTier:    defaultRange,
-		enterpriseTier: enterpriseRange,
-		highScaleTier:  highScaleRange,
-		zonalTier:      zonalRange,
-		premiumTier:    premiumRange,
-		basicSSDTier:   premiumRange, //these two are aliases
-		basicHDDTier:   defaultRange, //these two are aliases
-	}
-
+// provisionableCapacityForTier returns capacity range for tier
+func provisionableCapacityForTier(capRange *csi.CapacityRange, tier string) *capacityRangeForTier {
 	tier = strings.ToLower(tier)
-	validRange, ok := provisionableCapacityForTier[tier]
+	if tier == zonalTier && capRange != nil && capRange.GetRequiredBytes() > zonalLowTierMaxSize {
```

Comment:
consider a case where first the zonal high tier is created, then a zonal small tier is created, wouldn't that result in requested Capacity not valid?

Reply:
Pardon, I didn't get your question. Could you please elaborate? I have tested this flow and that works. These will be different requests right with different PVC name?

Reply:
This code appears a little buggy to me. We are updating the map with new entry every time and and immediately accessing it.

In that case let's not even create that entry in the map and evaluate the range on the fly depending on the zonal tier and band without caching it.

@leiyiz Is the CSI driver implementation thread safe in general? Can there be multiple create volume requests to the same driver? In that case this piece of code might not work as expected.

Reply:
yeah I think actually writing the map here is problematic. Fixed this by using the map for read only.

Reply:
> consider a case where first the zonal high tier is created, then a zonal small tier is created, wouldn't that result in requested Capacity not valid?

I see what you mean. Updated the code to not overwrite the map.